publish_topics:
    # to mc
    state: "/state"                                         # Vehicle State message.
    visualization: "/visualization"                         # Visualization message to the Master Control.
    connection: "/connection"                               # Connection State message.
    factsheet: "/factsheet"                                 # Setup of AGV in master control.
    # to agv
    order: "processed_order"                                # Processed order.
    instant_action: "processed_instant_action"              # Processed instant action message.
subscribe_topics:
    # from mc
    order_from_mc: "/order"                                 # Raw order coming from the cloud.
    ia_from_mc: "/instant_action"                           # Raw instant action message from the cloud.
    # from agv
    zone_set_id: "/zone_set_id"                             # ID of the current zone set used by the robot.                         !!! Uses ROS String messages. !!!
    order_state: "action_client/order_state"                # Current state of the running order. Contains the following fields :
                                                            # - orderId.
                                                            # - orderUpdateId
                                                            # - lastNodeId
                                                            # - lastNodeSequenceId
                                                            # - nodeStates
                                                            # - edgeStates
                                                            # - actionStates
    map_id: "/map_id"                                       # Current map id                                                        !!! Uses ROS String messages. !!!
    position_initialized: "/position_initialized"           # Boolean if the robot's position is initialized on the map.            !!! Uses ROS Bool messages. !!!
    localization_score: "/localization_score"               # Messages containing the vehicle's localization score.                 !!! Uses ROS Float64 messages. !!!
    pose: "/pose"                                           # Robot's pose on the map                                               !!! Uses ROS Pose messages. !!!
    velocity: "/velocity"                                   # Robot's velocity                                                      !!! Uses ROS Twist messages. !!!
    loads: "/loads"                                         # Loads carried by the robot
    paused: "/paused"                                       # Boolean if the robot is paused                                        !!! Uses ROS Bool messages. !!!
    new_base_request: "/new_base_request"                   # Boolean if the robot is running out of base nodes and 
                                                            # requires an order update                                              !!! Uses ROS Bool messages. !!!
    distance_since_last_node: "/distance_since_last_node"   # Distance traversed by the robot since the last node.
    battery_state: "/battery_state"                         # Battery state of the robot                                            !!! Uses ROS Battery messages. !!!
    operating_mode: "/operating_mode"                       # Robot's operating mode                                                !!! Uses ROS String messages. !!!
    errors: "/errors"                                       # Errors that occurred on the robot
    information: "/information"                             # Information messages from the robot
    safety_state: "/safety_state"                           # Robot's safety state
    interaction_zones: "/interaction_zones"                 # State of the interaction zones.

publish_periods:
    state_msg: 0.8                                          # Period on which to send state message if no new triggers
    visualization_msg: 0.3                                  # Period on which to send visualization message
    conn_msg: 15.0                                          # Period on which to send connection message